src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -0,0 +1,227 @@
+      gmio.put("burst_length_in_16byte",gmio_node.second.get<uint16_t>("burst_length_in_16byte"));
+      gmio.put("pl_port_name",gmio_node.second.get<std::string>("PL_port_name","N/A"));
+      gmio.put("pl_parameter_name",gmio_node.second.get<std::string>("PL_parameter_name","N/A"));
+      gmio_array.push_back(std::make_pair(gmio_node.first, gmio));
+    }
+    pt.add_child("gmios",gmio_array);
+
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+  }
+
+  return pt;
+}
+
+void
+ReportAie::getPropertyTreeInternal( const xrt_core::device * _pDevice, 
+                                              boost::property_tree::ptree &_pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data, 
+  // Then update this method to do so.
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void 
+ReportAie::getPropertyTree20202( const xrt_core::device * _pDevice, 
+                                           boost::property_tree::ptree &_pt) const
+{
+  _pt.add_child("aie_metadata", populate_aie(_pDevice, "Aie_Metadata"));
+}
+
+void 
+ReportAie::writeReport( const xrt_core::device * _pDevice,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & _output) const
+{
+  boost::property_tree::ptree _pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(_pDevice, _pt);
+
+  _output << "Aie\n";
+  _output << boost::format("  %-10s\n") % _pt.get<std::string>("aie_metadata.description");
+
+  try {
+    for (auto& gr: _pt.get_child("aie_metadata.graphs")) {
+      boost::property_tree::ptree& graph = gr.second;
+      _output << boost::format("  GRAPH[%2d] Name:%2s\n") % graph.get<std::string>("id")
+           % graph.get<std::string>("name");
+      _output << boost::format("    SNo.  %-20s%-30s%-30s\n") % "Core [C:R]"
+           % "Iteration_Memory [C:R]" % "Iteration_Memory_Addresses";
+      int count = 0;
+      for (auto& node : graph.get_child("tile")) {
+        boost::property_tree::ptree& tile = node.second;
+        _output << boost::format("    [%2d]   %-20s%-30s%-30d\n") % count
+            % (tile.get<std::string>("col")+":"+tile.get<std::string>("row"))
+            % (tile.get<std::string>("memcol")+":"+tile.get<std::string>("memrow"))
+            % node.second.get<uint16_t>("memaddr");
+        count++;
+      }
+      _output << std::endl;
+
+      _output << boost::format("    %s\n") % "Pl Kernel Instances in Graph:";
+      for (auto& node : graph.get_child("pl_kernel")) {
+        _output << boost::format("      %s\n") % node.second.data();
+      }
+      _output << std::endl;
+    }
+
+    int count = 0;
+    for (auto& rtp_node : _pt.get_child("aie_metadata.rtps")) {
+      _output << boost::format("  %-3s:[%2d]\n") % "RTP" % count;
+      _output << boost::format("    %-22s: %s\n") % "Port Name" % rtp_node.second.get<std::string>("port_name");

```

Comment:
Being that the same boost::format() is used multiple times, it would be best to first create the format:
`boost::format fmtCommon("  %-3s:[%2d]\n");`
and re-use it:
` _output << fmtCommon % "Port Name" % rtp_node.second.get<std::string>("port_name");`
That way if changes are needed, it only needs to be done in 1 location.